livekit-plugins/livekit-plugins-google/livekit/plugins/google/beta/realtime/realtime_api.py

```diff
@@ -498,84 +582,90 @@ def _start_new_generation(self):
             user_initiated=False,
         )
 
-        # Resolve any pending future from generate_reply()
-        if self._pending_generation_event_id and (
-            fut := self._response_created_futures.pop(self._pending_generation_event_id, None)
-        ):
-            fut.set_result(generation_event)
+        if self._pending_generation_fut and not self._pending_generation_fut.done():
+            generation_event.user_initiated = True
+            self._pending_generation_fut.set_result(generation_event)
```

Comment:
We need to check if this generation is actually the one we triggered or it is coming from another generation (like turn_detection)

I think the previous impl was using this? _pending_generation_event_id? Was it the intent?

Reply:
Yeah if the user speaks at the same time